Formula/openssl11.rb

```diff
@@ -0,0 +1,176 @@
+class Openssl11 < Formula
+  desc "Cryptography and SSL/TLS Toolkit"
+  homepage "https://openssl.org/"
+  url "https://www.openssl.org/source/openssl-1.1.0.tar.gz"
+  mirror "https://www.mirrorservice.org/sites/ftp.openssl.org/source/openssl-1.1.0.tar.gz"
+  sha256 "f5c69ff9ac1472c80b868efc1c1c0d8dcfc746d29ebe563de2365dd56dbd8c82"
+
+  keg_only :provided_by_osx,
+    "Apple has deprecated use of OpenSSL in favor of its own TLS and crypto libraries"
+
+  option :universal
+  option "without-test", "Skip build-time tests (not recommended)"
+
+  # Only needs 5.10 to run, but needs >5.13.4 to run the testsuite.
+  # https://github.com/openssl/openssl/blob/4b16fa791d3ad8/README.PERL
+  # The MacOS ML tag is same hack as the way we handle most :python deps.
+  if build.with? "test"
+    depends_on :perl => "5.14" if MacOS.version <= :mountain_lion
+  else
+    depends_on :perl => "5.10"
+  end
+
+  # This isn't technically needed where Clang exists, but is a small
+  # dep and is hard to reliably express non-Clang-only in dependency format.
+  depends_on "makedepend" => :build
```

Comment:
If you changed the default algorithms you needed to run `make depend` to rejiggle _(technical term that)_ things to build correctly before. I _think_ it would still be necessary if you've got a dirty source. Documentation upstream is kind of "eh" on the `makedepend` situation. There's a commit [here](https://github.com/openssl/openssl/commit/d20bb611d90df0dc3561a6972ba6bf0e3c65b3e5) that adds some detail, but not much.

I'll try poking it locally. If it breaks I guess we could always add the dep in later.